platforms/jvm/testing-junit-platform/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestClassProcessor.java

```diff
@@ -319,6 +321,16 @@ private Optional<String> className(TestDescriptor descriptor) {
     private static class BackwardsCompatibleLauncherSession implements AutoCloseable {
 
         static BackwardsCompatibleLauncherSession open() {
+            try {
+                Class.forName("org.junit.platform.launcher.core.LauncherFactory");
+            } catch (ClassNotFoundException e) {
+                throw new InvalidUserCodeException(
+                    "Failed to load JUnit Platform. " +
+                    "Please ensure that the JUnit Platform is available on the test runtime classpath. " +
+                    "See the user guide for more details: " + new DocumentationRegistry().getDocumentationFor("java_testing", "using_junit5")
+                );
```

Comment:
Can we keep the cause here?
```suggestion
                    "See the user guide for more details: " + new DocumentationRegistry().getDocumentationFor("java_testing", "using_junit5"),
                    e
                );
```

Reply:
I had this originally but decided to remove it.
The cause is a ClassNotFoundException, I don't think we should be presenting that to the user, as we have identified the solution to the problem already

Reply:
Makes sense. I suppose it's probably not too necessary to debug the issue.